src/BaGetter/Authentication/NugetBasicAuthenticationHandler.cs

```diff
@@ -0,0 +1,91 @@
+using Azure.Core;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using BaGetter.Core;
+using System.Linq;
+
+namespace BaGetter.Web.Authentication;
+
+public class NugetBasicAuthenticationHandler : AuthenticationHandler<AuthenticationSchemeOptions>
+{
+    private readonly IOptions<BaGetterOptions> bagetterOptions;
+
+    public NugetBasicAuthenticationHandler(
+        IOptionsMonitor<AuthenticationSchemeOptions> options,
+        ILoggerFactory logger,
+        UrlEncoder encoder,
+        IOptions<BaGetterOptions> bagetterOptions)
+        : base(options, logger, encoder)
+    {
+        this.bagetterOptions = bagetterOptions;
+    }
+
+    protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        if (bagetterOptions.Value.Authentication is null || bagetterOptions.Value.Authentication.Length==0 || bagetterOptions.Value.Authentication.All(a=> string.IsNullOrWhiteSpace(a.Username) && string.IsNullOrWhiteSpace(a.Password)))
+        {
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Anonymous, string.Empty),
+            };
+            var identity = new ClaimsIdentity(claims, Scheme.Name);
+            var principal = new ClaimsPrincipal(identity);
+
+            var ticket = new AuthenticationTicket(principal, Scheme.Name);
+
+            return Task.FromResult(AuthenticateResult.Success(ticket));
```

Comment:
I'd prefer a private method to wrap this. Not sure what to call it, maybe `CreateAnonymousAuthClaim()`? `CreateAnonymousLogin()`? `AcceptAnonymousAuthentication()`? I'd probably consult chatgpt for this one but maybe you have a better idea.